### PR TITLE
#432 Selection of factory method based on Qualifier

### PR DESCRIPTION
**Pull request by sjaakd**
* Aligning `@BeanMapping` with other mapping types.
* Small improvement on templates (removing empty lines from fields)

### CODE REVIEW

File: processor/src/main/java/org/mapstruct/ap/model/source/BeanMapping.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public static BeanMapping fromPrism( BeanMappingPrism beanMapping, ExecutableElement method, Messager messager ) {
+        if ( beanMapping == null ) {
+            return null;
+        }
+
+        boolean resultTypeIsDefined = !TypeKind.VOID.equals( beanMapping.resultType().getKind() );
+        if ( !resultTypeIsDefined && beanMapping.qualifiedBy().isEmpty() ) {
+            messager.printMessage(
+                Diagnostic.Kind.ERROR,
+                "'resultType' and 'qualifiedBy' are are are undefined in @BeanMapping, "
```
Comment (gunnarmorling) [inline processor/src/main/java/org/mapstruct/ap/model/source/BeanMapping.java]: Some "are" too much here :)

File: core-common/src/main/java/org/mapstruct/BeanMapping.java [hunk 2]
```diff
@@ -38,4 +39,15 @@
...
+   /**
+     * A qualifier can be specified to aid the selection process of a suitable factory method. This is useful in
+     * case multiple factory method (hand written of internal) qualify and result in an 'Ambiguous factory methods'
+     * error.
+     *
+     * A qualifier is a custom annotation and can be placed on either a hand written mapper class or a method.
+     *
+     * @return the qualifiers
+     */
+    Class<? extends Annotation>[] qualifiedBy() default { };
```
Comment (gunnarmorling) [inline core-common/src/main/java/org/mapstruct/BeanMapping.java]: Can we add a note that either this or the other attribute is to be specified. Similar to how it's done for `@Mapping`.

**Comment by gunnarmorling** [comment]
Looks good to me.

**Comment by sjaakd** [comment]
@gunnarmorling  thanks for reviewing. I processed your comments and will merge soon.

**Pull request merged**